### PR TITLE
SONAR-9245 Return leak period in api/components/search_projects

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: it/it-tests/src/test/java/it/projectSearch/SearchProjectsTest.java [hunk 1]
```diff
@@ -223,7 +227,35 @@ public void provisioned_projects_should_be_included_to_results() throws Exceptio
...
 
-    assertThat(response.getComponentsList()).extracting(Component::getKey).contains(projectKey);
+    assertThat(response.getComponentsList()).extracting(Component::getKey).containsOnly(projectKey);
+  }
+
+  @Test
+  public void return_leak_period_date() throws Exception {
+    setServerProperty(orchestrator, "sonar.leak.period", "previous_version");
+    Date leakPeriodDate = toDate("2016-12-31");
+    // This project has no leak period
```
Comment (bartfastiel) [inline it/it-tests/src/test/java/it/projectSearch/SearchProjectsTest.java]: typo: `no` => `a`

File: it/it-tests/src/test/java/it/projectSearch/SearchProjectsTest.java [hunk 2]
```diff
@@ -223,7 +227,35 @@ public void provisioned_projects_should_be_included_to_results() throws Exceptio
...
+      SearchProjectsRequest.builder().setAdditionalFields(singletonList("leakPeriodDate")).setOrganization(organizationKey).build());
+
+    assertThat(response.getComponentsList()).extracting(Component::getKey, Component::hasLeakPeriodDate)
+      .containsOnly(
+        tuple(projectKey1, true),
+        tuple(projectKey2, false),
+        tuple(projectKey3, false));
+    Component project1 = response.getComponentsList().stream().filter(component -> component.getKey().equals(projectKey1)).findFirst()
+      .orElseThrow(() -> new IllegalStateException("Project1 is not found"));
+    assertThat(project1.getLeakPeriodDate()).startsWith("2016-12-31");
```
Comment (bartfastiel) [inline it/it-tests/src/test/java/it/projectSearch/SearchProjectsTest.java]: Would `isEqualTo` be possible here?

File: server/sonar-server/src/test/java/org/sonar/server/component/ws/SearchProjectsActionTest.java [hunk 3]
```diff
@@ -150,8 +150,8 @@ public void verify_definition() {
     assertThat(def.changelog()).extracting(Change::getVersion, Change::getDescription).containsExactlyInAnyOrder(
       tuple("6.4", "The 'languages' parameter accepts 'filter' to filter by language"),
       tuple("6.4", "The 'visibility' field is added"),
-      tuple("6.5", "The 'filter' parameter now allows 'NO_DATA' as value for numeric metrics")
-    );
+      tuple("6.5", "The 'filter' parameter now allows 'NO_DATA' as value for numeric metrics"),
+      tuple("6.5", "Value 'leakPeriodDate' is added to parameter '[analysisDate, leakPeriodDate]'"));
```
Comment (bartfastiel) [inline server/sonar-server/src/test/java/org/sonar/server/component/ws/SearchProjectsActionTest.java]: I suggest to change this to `Value 'leakPeriodDate' is added to parameter 'f'`.

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchProjectsAction.java [hunk 4]
```diff
@@ -390,14 +393,17 @@ private BucketToFacetValue() {
...
     private final Component.Builder wsComponent;
     private final Map<String, OrganizationDto> organizationsByUuid;
     private final Set<String> favoriteProjectUuids;
     private final boolean isUserLoggedIn;
     private final Map<String, SnapshotDto> analysisByProjectUuid;
 
-    private DbToWsComponent(Map<String, OrganizationDto> organizationsByUuid, Set<String> favoriteProjectUuids, Map<String, SnapshotDto> analysisByProjectUuid,
+    private DbToWsComponent(SearchProjectsRequest request, Map<String, OrganizationDto> organizationsByUuid, Set<String> favoriteProjectUuids,
+      Map<String, SnapshotDto> analysisByProjectUuid,
       boolean isUserLoggedIn) {
```
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchProjectsAction.java]: I would prefer one line break less here.

File: server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchProjectsAction.java [hunk 5]
```diff
@@ -408,9 +414,7 @@ private DbToWsComponent(Map<String, OrganizationDto> organizationsByUuid, Set<St
     @Override
     public Component apply(ComponentDto dbComponent) {
       OrganizationDto organizationDto = organizationsByUuid.get(dbComponent.getOrganizationUuid());
-      if (organizationDto == null) {
-        throw new NotFoundException(format("Organization with uuid '%s' not found", dbComponent.getOrganizationUuid()));
-      }
+      checkFound(organizationDto, "Organization with uuid '%s' not found", dbComponent.getOrganizationUuid());
```
Comment (bartfastiel) [inline server/sonar-server/src/main/java/org/sonar/server/component/ws/SearchProjectsAction.java]: `dbComponent.getOrganizationUuid()` could be externalised to a variable

File: it/it-tests/src/test/java/it/projectSearch/SearchProjectsTest.java [hunk 6]
```diff
@@ -212,7 +216,35 @@ public void provisioned_projects_should_be_included_to_results() throws Exceptio
...
     SearchProjectsWsResponse response = searchProjects(SearchProjectsRequest.builder().setOrganization(organizationKey).build());
 
-    assertThat(response.getComponentsList()).extracting(Component::getKey).contains(projectKey);
+    assertThat(response.getComponentsList()).extracting(Component::getKey).containsOnly(projectKey);
+  }
+
+  @Test
+  public void return_leak_period_date() throws Exception {
+    setServerProperty(orchestrator, "sonar.leak.period", "previous_version");
+    Date leakPeriodDate = toDate("2016-12-31");
```
Comment (julienlancelot) [inline it/it-tests/src/test/java/it/projectSearch/SearchProjectsTest.java]: Nope indeed

File: it/it-tests/src/test/java/it/projectSearch/SearchProjectsTest.java (hunk 2 above)
Comment (julienlancelot) [inline it/it-tests/src/test/java/it/projectSearch/SearchProjectsTest.java]: The date from the WS contains the hour

**Pull request merged**